### DISCUSSION TITLE
Getting the node from the server and adding it to the ReactFlow state

### DISCUSSION THREAD

**Discussion by 0xLDev** [xyflow/xyflow#2748] [Q&A]
I have a state in Redux Toolkit, when I make a request to the server, it returns to me the nodes how I rightfully print them to display, tried through useNodesState does not work gives an empty array, but when I pull the array from the state to the console log it is not empty

**Comment by moklick** [comment]
When you are using Redux, you should keep your nodes in your Redux store. There is no need to use `useNodesState` (you only use that if you want to keep the state in your component). We have a guide about how to use a state management lib with React Flow: https://reactflow.dev/docs/guides/state-management/ It uses Zustand instead of Redux but the concepts are quite similar.
